src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import {
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiContextMenu,
+  EuiNotificationBadge,
+  EuiPopover,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { SavedObjectsClientContract, ToastsStart } from 'opensearch-dashboards/public';
+import { DataSourceOption } from '../data_source_selector/data_source_selector';
+import { getDataSourcesWithFields } from '../utils';
+
+interface DataSourceAggregatedViewProps {
+  savedObjectsClient: SavedObjectsClientContract;
+  notifications: ToastsStart;
+  hideLocalCluster: boolean;
+  fullWidth: boolean;
+  activeDatasourceIds?: string[];
+  filterFn?: (dataSource: any) => boolean;
+  displayAllCompatibleDataSources: boolean;
+}
+
+interface DataSourceAggregatedViewState {
+  dataSourceOptions: DataSourceOption[];
+  selectedOption: DataSourceOption[];
+  isPopoverOpen: boolean;
+  allDataSourcesIdMap: Map<string, any>;
+}
+
+export class DataSourceAggregatedView extends React.Component<
+  DataSourceAggregatedViewProps,
+  DataSourceAggregatedViewState
+> {
+  private _isMounted: boolean = false;
+
+  constructor(props: DataSourceAggregatedViewProps) {
+    super(props);
+
+    this.state = {
```

Comment:
I noticed a typescript error when checking the code locally, not sure if it's just my editor's issue:
```
Type '{ isPopoverOpen: false; allDataSourcesIdMap: Map<any, any>; }' is missing the following properties from type 'Readonly<DataSourceAggregatedViewState>': dataSourceOptions, selectedOption (tsserver 2739)
```

Reply:
Thanks! The 2 fields are not needed anymore, and cleaned up in new commit